<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>Crohn&#39;s VS Ulcerative Colitis</title>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>

<!-- MathJax scripts -->
<script type="text/javascript" src="https://c328740.ssl.cf1.rackcdn.com/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>


<style type="text/css">
body {
	box-sizing: border-box;
	font-size: 16px;
	font-family: 'Myriad Pro', Calibri, Helvetica, Arial, sans-serif;
	line-height: 1.5;
	-webkit-font-smoothing: antialiased;
	width:800px;
	margin: auto;
}

h1, h2, h3, h4, h5, h6 {
	margin: 10px 0;
	font-weight: 700;
	color:#222222;
		font-family: 'Lucida Grande', 'Calibri', Helvetica, Arial, sans-serif;
	letter-spacing: -1px;
}

h1 {
	font-size: 36px;
	font-weight: 700;
}

h2 {
	padding-bottom: 10px;
	font-size: 32px;
	background: url('../images/bg_hr.png') repeat-x bottom;
}

h3 {
	font-size: 24px;
}

h4 {
	font-size: 21px;
}

h5 {
	font-size: 18px;
}

h6 {
	font-size: 16px;
}

p {
	margin: 10px 0 15px 0;
	text-align: justify;
}

tt, code, pre {
	font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}



a:visited {
	color: rgb(50%, 0%, 50%);
}

pre {	
	margin-top: 0;
	max-width: 95%;
	border: 1px solid #ccc;
}

pre code {
	display: block; padding: 0.5em;
}

code.r {
	background-color: #F8F8F8;
}

table, td, th {
	border: none;
	border-spacing:20px; 5px;
}

blockquote {
	color:#666666;
		margin:0;
	padding-left: 1em;
	border-left: 0.5em #EEE solid;
}

hr {
	height: 0px;
	border-bottom: none;
	border-top-width: thin;
	border-top-style: dotted;
	border-top-color: #999999;
}

@media print {
	* { 
		background: transparent !important; 
		color: black !important; 
		filter:none !important; 
		-ms-filter: none !important; 
	}
	
	body { 
		font-size:12pt; 
		max-width:100%; 
	}
	
	a, a:visited { 
		text-decoration: underline; 
	}
	
	hr { 
		visibility: hidden;
		page-break-before: always;
	}
	
	pre, blockquote { 
		padding-right: 1em; 
		page-break-inside: avoid; 
	}
	
	tr, img { 
		page-break-inside: avoid; 
	}
	
	img { 
		max-width: 100% !important; 
	}
	
	@page :left { 
		margin: 15mm 20mm 15mm 10mm; 
	}
	
	@page :right { 
		margin: 15mm 10mm 15mm 20mm; 
	}
	
	p, h2, h3 { 
		orphans: 3; widows: 3; 
	}
	
	h2, h3 { 
		page-break-after: avoid; 
	}
}
</style>



</head>

<body>
<!--
%\VignetteEngine{knitr}
%\VignetteIndexEntry{UC vs Crohn's}
%\VignetteDepends{ccPaper, GEOquery, limma, hgu133plus2.db, GO.db, org.Hs.eg.db}
-->

<pre><code class="r">library(ccPaper)
</code></pre>

<h1>Crohn&#39;s VS Ulcerative Colitis</h1>

<p>GEO dataset <strong>GSE36807</strong> has crohn&#39;s, uc, and normal samples. </p>

<pre><code class="r">library(GEOquery)

cucData &lt;- getGEO(GEO = &quot;GSE36807&quot;)[[1]]

tmpP &lt;- pData(cucData)
tmpP$status &lt;- &quot;control&quot;

cLoc &lt;- grep(&quot;Crohn&quot;, tmpP$characteristics_ch1)
tmpP$status[cLoc] &lt;- &quot;crohns&quot;

ucLoc &lt;- grep(&quot;Ulcerative colitis&quot;, tmpP$characteristics_ch1)
tmpP$status[ucLoc] &lt;- &quot;uc&quot;
pData(cucData) &lt;- tmpP

.sessionInfo &lt;- sessionInfo()
.timeDate &lt;- Sys.time()
save(cucData, .sessionInfo, .timeDate, file = &quot;inst/data/uc_crohns_rawData.RData&quot;)
</code></pre>

<h2>Differential analysis</h2>

<p>Use the <code>limma</code> package to rank probes. </p>

<pre><code class="r">library(limma)
library(hgu133plus2.db)
data(uc_crohns_rawData)

cucComps &lt;- c(&quot;uc - control&quot;, &quot;crohns - control&quot;)
geneID &lt;- unlist(mget(featureNames(cucData), hgu133plus2ENTREZID))

cucExpr &lt;- exprs(cucData)
cucCollapse &lt;- collapseProbes(cucExpr, geneID)  # collapse to single genes using median of expression
cucCharacter &lt;- pData(cucData)

cucFC &lt;- rankGenes(cucCollapse, cucCharacter$status, cucComps, doAggregation = FALSE, 
    aggregateIndex = NA)
names(cucFC) &lt;- c(&quot;UC&quot;, &quot;CROHNS&quot;)

cucDiff &lt;- lapply(cucFC, getDiffGenes, id = &quot;id&quot;)
# this object can be run using base ccEnrich, or can be easily coerced
</code></pre>

<h2>ORA</h2>

<p>Do normal over-representation analysis on the <code>cucDiff</code>, those genes differentially expressed in the Crohn&#39;s and UC samples.</p>

<pre><code class="r">library(GO.db)
cucGeneList &lt;- list(UC_up = list(genes = cucDiff$UC$up, universe = cucDiff$UC$universe, 
    annotation = &quot;org.Hs.eg.db&quot;), UC_dn = list(genes = cucDiff$UC$dn, universe = cucDiff$UC$universe, 
    annotation = &quot;org.Hs.eg.db&quot;), CR_up = list(genes = cucDiff$CROHNS$up, universe = cucDiff$CROHNS$universe, 
    annotation = &quot;org.Hs.eg.db&quot;), CR_dn = list(genes = cucDiff$CROHNS$dn, universe = cucDiff$CROHNS$universe, 
    annotation = &quot;org.Hs.eg.db&quot;))
cucGeneList &lt;- new(&quot;ccGeneList&quot;, cucGeneList, ccType = &quot;BP&quot;)
</code></pre>

<pre><code>## Warning: NAs introduced by coercion
</code></pre>

<pre><code class="r">
cucEnrich &lt;- ccEnrich(cucGeneList)
</code></pre>

<pre><code>## Performing Enrichment Calculations ....
## UC_up : BP 
## UC_dn : BP 
## CR_up : BP 
## CR_dn : BP 
## Done!!
</code></pre>

<pre><code class="r">pvalueType(cucEnrich) &lt;- &quot;pval&quot;
pvalueCutoff(cucEnrich) &lt;- 0.01

cucEnrich
</code></pre>

<pre><code>##     Annotation category:  GO   BP 
##                FDR runs:  50 
## Default p-values to use:  pval 
##                 pCutoff:  0.01 
## 
## List:  UC_up 
## Gene to GO BP  test for over-representation 
## 6369 GO BP ids tested (657 have p &lt;= 0.01 &amp; count &gt;= 0)
## Selected gene set size: 1207 
##     Gene universe size: 13833 
##     Annotation package: org.Hs.eg 
## 
## List:  UC_dn 
## Gene to GO BP  test for over-representation 
## 5621 GO BP ids tested (218 have p &lt;= 0.01 &amp; count &gt;= 0)
## Selected gene set size: 935 
##     Gene universe size: 13833 
##     Annotation package: org.Hs.eg 
## 
## List:  CR_up 
## Gene to GO BP  test for over-representation 
## 1675 GO BP ids tested (179 have p &lt;= 0.01 &amp; count &gt;= 0)
## Selected gene set size: 48 
##     Gene universe size: 13833 
##     Annotation package: org.Hs.eg 
## 
## List:  CR_dn 
## Gene to GO BP  test for over-representation 
## 2132 GO BP ids tested (70 have p &lt;= 0.01 &amp; count &gt;= 0)
## Selected gene set size: 116 
##     Gene universe size: 13833 
##     Annotation package: org.Hs.eg
</code></pre>

<pre><code class="r">
cucOpts &lt;- new(&quot;ccOptions&quot;, listNames = names(cucGeneList), compareNames = c(&quot;UC_up&quot;, 
    &quot;UC_dn&quot;, &quot;CR_up&quot;, &quot;CR_dn&quot;, &quot;UC_up,CR_up&quot;, &quot;UC_up,CR_dn&quot;, &quot;UC_dn,CR_up&quot;, 
    &quot;UC_dn,CR_dn&quot;))
cucCompare &lt;- ccCompare(cucEnrich, cucOpts)

cucCompare
</code></pre>

<pre><code>## ccCompare results for:
## 
## Annotation category:  GO   BP 
## Main graph: A graphNEL graph with directed edges
## Number of Nodes = 621 
## Number of Edges = 117418
</code></pre>

<pre><code class="r">cucCompare &lt;- breakEdges(cucCompare$BP, 0.8)
cucCy &lt;- ccOutCyt(cucCompare, cucOpts)
breakEdges(cucCy, 1)
</code></pre>

<h2>Run GSEA type analysis using limma</h2>

<p>There are two ways to do GSEA using <code>limma</code>. </p>

<p>1 Use <code>romer</code> for competitive tests with rotations to do random samples
  2 Use <code>geneSetTest</code> for gene-wise competitive tests, <em>i.e.</em> each set is tested against random gene samples</p>

<p>We&#39;re going to try both and see what we get out.</p>

<pre><code class="r">library(GO.db)
library(org.Hs.eg.db)
library(limma)
hsGO &lt;- as.list(org.Hs.egGO2ALLEGS)
hsGO &lt;- hsGO[(Ontology(names(hsGO)) == &quot;BP&quot;)]
hsGO &lt;- lapply(hsGO, unique)

useSet &lt;- symbols2indices(hsGO, rownames(cucCollapse))
</code></pre>

<p>Run using <code>romer</code></p>

<pre><code class="r">sampleStatus &lt;- cucCharacter$status
doComps &lt;- cucComps
f &lt;- factor(sampleStatus)
design &lt;- model.matrix(~0 + f)
colnames(design) &lt;- levels(f)

contrast.matrix &lt;- makeContrasts(contrasts = doComps, levels = design)
</code></pre>

<pre><code class="r">cucRomer &lt;- multicontrastRomer(useSet, cucCollapse, design, contrast.matrix, 
    nrot = 10000)
names(cucRomer) &lt;- c(&quot;UC&quot;, &quot;CROHNS&quot;)
save(cucRomer, file = &quot;inst/data/cucRomer.RData&quot;)

</code></pre>

<pre><code class="r">data(cucRomer)

geneAnnMapping &lt;- new(&quot;namedList&quot;, .Data = hsGO, names = names(hsGO))

getSigID &lt;- function(inRomer, pCut = 0.05, whichCol = c(&quot;Up&quot;, &quot;Down&quot;)) {
    sigID &lt;- lapply(whichCol, function(inCol) {
        rownames(inRomer[(inRomer[, inCol] &lt;= pCut), ])
    })
    names(sigID) &lt;- whichCol
    return(sigID)
}

cucSigRomer &lt;- lapply(cucRomer, getSigID, pCut = 0.01)
cucSigRomer &lt;- unlist(cucSigRomer, recursive = FALSE)

genCCSigList &lt;- function(inSig) {
    tmp &lt;- new(&quot;ccSigList&quot;, sigID = inSig)
}

cucCCSig &lt;- lapply(cucSigRomer, genCCSigList)

cucCCRomer &lt;- new(&quot;GENccEnrichResult&quot;, cucCCSig, categoryName = &quot;GSEAGO&quot;, geneAnnMapping = geneAnnMapping, 
    overlapType = &quot;overlap&quot;, annDescription = Term(names(geneAnnMapping)))

cucRomerOpts &lt;- new(&quot;ccOptions&quot;, listNames = names(cucCCRomer), compareNames = c(&quot;UC.Up&quot;, 
    &quot;UC.Down&quot;, &quot;CROHNS.Up&quot;, &quot;CROHNS.Down&quot;, &quot;UC.Up,CROHNS.Up&quot;, &quot;UC.Up,CROHNS.Down&quot;, 
    &quot;UC.Down,CROHNS.Up&quot;, &quot;UC.Down,CROHNS.Down&quot;))
compareCUCRomer &lt;- ccCompare(cucCCRomer, cucRomerOpts)
compareCUCRomer
</code></pre>

<pre><code>## Annotation category:  GSEAGO    
## Main graph: A graphNEL graph with directed edges
## Number of Nodes = 288 
## Number of Edges = 15342
</code></pre>

<pre><code class="r">compareCUCRomer &lt;- breakEdges(compareCUCRomer, 0.8)
cucRomerCW &lt;- ccOutCyt(compareCUCRomer, cucRomerOpts, postText = &quot;romer&quot;, rpcPort = 9001)
generateLegend(cucRomerOpts)
</code></pre>

<p>Now we will save groups so that they can be queried and examined later.</p>

<pre><code class="r">cucNodeGroups &lt;- cytOutNodes(&quot;response to lipopolysaccharide and bacterial - UC.Up&quot;, 
    cucRomerCW)
cucNodeGroups &lt;- cytOutNodes(&quot;regulation of inflammatory response - UC.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;hydrogen peroxide metabolism - UC.Up,CROHNS.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;regulation of cell cycle and DNA damage response - UC.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;regulation of ubiquitination and ligase activity - UC.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;nucleotide and nucleoside metabolism - UC.Up,CROHNS.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;amine metabolism - UC.Up,CROHNS.Up&quot;, cucRomerCW, 
    cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;glandular cell differentiation - UC.Down&quot;, cucRomerCW, 
    cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;oligodendrocyte differentiation - UC.Down,CROHNS.Down&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;cellular pattern specification - UC.Down,CROHNS.Down&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;NAD biosynthesis - CROHNS.Up&quot;, cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;hormone metabolism - CROHNS.Up&quot;, cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;extrinsic signal transduction - UC.Up,CROHNS.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;regulation of nitric-oxide synthase - UC.Up,CROHNS.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;fatty-acyl-CoA biosynthesis - UC.Up,CROHNS.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;response to growth hormone - CROHNS.Up&quot;, cucRomerCW, 
    cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;melanin metabolism - CROHNS.Up&quot;, cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;protein dephosphorylation - CROHNS.Up&quot;, cucRomerCW, 
    cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;chemokine and cytokine production - UC.Up,CROHNS.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;membrane biogenesis and assembly - UC.Down&quot;, cucRomerCW, 
    cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;activin receptor signaling - UC.Down&quot;, cucRomerCW, 
    cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;nik/nk-kappab cascade - UC.Up&quot;, cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;er unfolded protein response - UC.Up, CROHNS.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;regulation of ras/rac/rho gtpase activity - UC.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;antigen processing and presentation - UC.UP,CROHNS.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;COPII vesicle coating and targeting - UC.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;negative regulation of peptidase activity - UC.Up&quot;, 
    cucRomerCW, cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;response to type 1 interferon - UC.Up&quot;, cucRomerCW, 
    cucNodeGroups)
cucNodeGroups &lt;- cytOutNodes(&quot;protein N-linked glycosylation - UC.Up&quot;, cucRomerCW, 
    cucNodeGroups)
.sessionInfo &lt;- sessionInfo()
.timeDate &lt;- Sys.time()
save(cucRomerCW, cucNodeGroups, compareCUCRomer, cucRomerOpts, .sessionInfo, 
    .timeDate, file = &quot;inst/data/cucCCOutput.RData&quot;)
</code></pre>

<p>We can then write out the results into a nice table.</p>

<pre><code class="r">data(cucCCOutput)
allDescStrings &lt;- sapply(cucNodeGroups, function(x) {
    x$descStr
})
string2List &lt;- strsplit(allDescStrings, &quot; - &quot;, fixed = TRUE)
justDesc &lt;- sapply(string2List, function(x) {
    x[1]
})
listMem &lt;- sapply(string2List, function(x) {
    x[2]
})
listMemSplit &lt;- strsplit(listMem, &quot;,&quot;, fixed = TRUE)

descMembershipTable &lt;- matrix(&quot;&quot;, nrow = length(allDescStrings), ncol = 5)
colnames(descMembershipTable) &lt;- c(&quot;Description&quot;, &quot;UC.Down&quot;, &quot;UC.Up&quot;, &quot;CROHNS.Down&quot;, 
    &quot;CROHNS.Up&quot;)

descMembershipTable[, &quot;Description&quot;] &lt;- justDesc

for (inRow in seq(1, nrow(descMembershipTable))) {
    useSplit &lt;- listMemSplit[[inRow]]
    trimSplit &lt;- gsub(&quot; &quot;, &quot;&quot;, useSplit)
    useLocs &lt;- sapply(trimSplit, grep, colnames(descMembershipTable), ignore.case = TRUE)
    descMembershipTable[inRow, useLocs] &lt;- &quot;X&quot;
}

orderBy &lt;- c(&quot;UC.Up,CROHNS.Up&quot;, &quot;UC.UP,CROHNS.Up&quot;, &quot;UC.Up&quot;, &quot;UC.Down&quot;, &quot;UC.Down,CROHNS.Down&quot;, 
    &quot;CROHNS.Up&quot;, &quot;CROHNS.Down&quot;)

listMem &lt;- gsub(&quot; &quot;, &quot;&quot;, listMem)
newOrder &lt;- unlist(lapply(orderBy, function(x) {
    which(listMem %in% x)
}))

descMembershipTable &lt;- descMembershipTable[newOrder, ]
require(xtable)

# add an html link to each entry in the table
useLink &lt;- paste(&quot;&lt;a href=\&quot;#loc&quot;, seq(1, nrow(descMembershipTable)), &quot;\&quot;&gt;&quot;, 
    descMembershipTable[, &quot;Description&quot;], &quot;&lt;/a&gt;&quot;, sep = &quot;&quot;)
descMembershipTable[, &quot;Description&quot;] &lt;- useLink
</code></pre>

<p>Print out the table.</p>

<p><a name="tableLink"></a></p>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:13 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> <a href="#loc1">hydrogen peroxide metabolism</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc2">nucleotide and nucleoside metabolism</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc3">amine metabolism</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc4">extrinsic signal transduction</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc5">regulation of nitric-oxide synthase</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc6">fatty-acyl-CoA biosynthesis</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc7">chemokine and cytokine production</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc8">er unfolded protein response</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc9">antigen processing and presentation</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc10">response to lipopolysaccharide and bacterial</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc11">regulation of inflammatory response</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc12">regulation of cell cycle and DNA damage response</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc13">regulation of ubiquitination and ligase activity</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc14">nik/nk-kappab cascade</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc15">regulation of ras/rac/rho gtpase activity</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc16">COPII vesicle coating and targeting</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc17">negative regulation of peptidase activity</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc18">response to type 1 interferon</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc19">protein N-linked glycosylation</a> </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc20">glandular cell differentiation</a> </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc21">membrane biogenesis and assembly</a> </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc22">activin receptor signaling</a> </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc23">oligodendrocyte differentiation</a> </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc24">cellular pattern specification</a> </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> </TR>
  <TR> <TD> <a href="#loc25">NAD biosynthesis</a> </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc26">hormone metabolism</a> </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc27">response to growth hormone</a> </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc28">melanin metabolism</a> </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> <a href="#loc29">protein dephosphorylation</a> </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p>We also want the set of GO terms, their descriptions, and list memberships so that the inquisitive mind can see how we generated the groups.</p>

<p><a name="loc1"></a> </p>

<h4>hydrogen peroxide metabolism - UC.Up,CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:13 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:2000377 </TD> <TD> regulation of reactive oxygen species metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0042554 </TD> <TD> superoxide anion generation </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0072593 </TD> <TD> reactive oxygen species metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0032928 </TD> <TD> regulation of superoxide anion generation </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0042743 </TD> <TD> hydrogen peroxide metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:2000379 </TD> <TD> positive regulation of reactive oxygen species metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc2"></a> </p>

<h4>nucleotide and nucleoside metabolism - UC.Up,CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:13 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0043094 </TD> <TD> cellular metabolic compound salvage </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0043174 </TD> <TD> nucleoside salvage </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0072529 </TD> <TD> pyrimidine-containing compound catabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0009112 </TD> <TD> nucleobase metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0072527 </TD> <TD> pyrimidine-containing compound metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0009113 </TD> <TD> purine nucleobase biosynthetic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0006213 </TD> <TD> pyrimidine nucleoside metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0043173 </TD> <TD> nucleotide salvage </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0008655 </TD> <TD> pyrimidine-containing compound salvage </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc3"></a> </p>

<h4>amine metabolism - UC.Up,CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:13 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0006596 </TD> <TD> polyamine biosynthetic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0033238 </TD> <TD> regulation of cellular amine metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0009308 </TD> <TD> amine metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0009309 </TD> <TD> amine biosynthetic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0006521 </TD> <TD> regulation of cellular amino acid metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0042401 </TD> <TD> cellular biogenic amine biosynthetic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0044106 </TD> <TD> cellular amine metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0033240 </TD> <TD> positive regulation of cellular amine metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc4"></a> </p>

<h4>extrinsic signal transduction - UC.Up,CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:13 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:2001240 </TD> <TD> negative regulation of extrinsic apoptotic signaling pathway in absence of ligand </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:1901099 </TD> <TD> negative regulation of signal transduction in absence of ligand </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:2001239 </TD> <TD> regulation of extrinsic apoptotic signaling pathway in absence of ligand </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0038034 </TD> <TD> signal transduction in absence of ligand </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0097192 </TD> <TD> extrinsic apoptotic signaling pathway in absence of ligand </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc5"></a> </p>

<h4>regulation of nitric-oxide synthase - UC.Up,CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:13 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0032770 </TD> <TD> positive regulation of monooxygenase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0051000 </TD> <TD> positive regulation of nitric-oxide synthase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0032768 </TD> <TD> regulation of monooxygenase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0050999 </TD> <TD> regulation of nitric-oxide synthase activity </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc6"></a> </p>

<h4>fatty-acyl-CoA biosynthesis - UC.Up,CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:13 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0035336 </TD> <TD> long-chain fatty-acyl-CoA metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0046949 </TD> <TD> fatty-acyl-CoA biosynthetic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0035338 </TD> <TD> long-chain fatty-acyl-CoA biosynthetic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0035337 </TD> <TD> fatty-acyl-CoA metabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc7"></a> </p>

<h4>chemokine and cytokine production - UC.Up,CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0032602 </TD> <TD> chemokine production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0009306 </TD> <TD> protein secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050704 </TD> <TD> regulation of interleukin-1 secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050707 </TD> <TD> regulation of cytokine secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050718 </TD> <TD> positive regulation of interleukin-1 beta secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032635 </TD> <TD> interleukin-6 production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050706 </TD> <TD> regulation of interleukin-1 beta secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032680 </TD> <TD> regulation of tumor necrosis factor production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050714 </TD> <TD> positive regulation of protein secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050715 </TD> <TD> positive regulation of cytokine secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050663 </TD> <TD> cytokine secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0042092 </TD> <TD> type 2 immune response </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0001818 </TD> <TD> negative regulation of cytokine production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032732 </TD> <TD> positive regulation of interleukin-1 production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050708 </TD> <TD> regulation of protein secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032651 </TD> <TD> regulation of interleukin-1 beta production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032755 </TD> <TD> positive regulation of interleukin-6 production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032760 </TD> <TD> positive regulation of tumor necrosis factor production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0001817 </TD> <TD> regulation of cytokine production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002643 </TD> <TD> regulation of tolerance induction </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032652 </TD> <TD> regulation of interleukin-1 production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032640 </TD> <TD> tumor necrosis factor production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032757 </TD> <TD> positive regulation of interleukin-8 production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032715 </TD> <TD> negative regulation of interleukin-6 production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032675 </TD> <TD> regulation of interleukin-6 production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0071706 </TD> <TD> tumor necrosis factor superfamily cytokine production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032642 </TD> <TD> regulation of chemokine production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032722 </TD> <TD> positive regulation of chemokine production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050716 </TD> <TD> positive regulation of interleukin-1 secretion </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032731 </TD> <TD> positive regulation of interleukin-1 beta production </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0042033 </TD> <TD> chemokine biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0050755 </TD> <TD> chemokine metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0045408 </TD> <TD> regulation of interleukin-6 biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc8"></a> </p>

<h4>er unfolded protein response - UC.Up, CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0070059 </TD> <TD> intrinsic apoptotic signaling pathway in response to endoplasmic reticulum stress </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0035966 </TD> <TD> response to topologically incorrect protein </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0035967 </TD> <TD> cellular response to topologically incorrect protein </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0034620 </TD> <TD> cellular response to unfolded protein </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032075 </TD> <TD> positive regulation of nuclease activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0006986 </TD> <TD> response to unfolded protein </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032069 </TD> <TD> regulation of nuclease activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0006987 </TD> <TD> activation of signaling protein activity involved in unfolded protein response </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0034976 </TD> <TD> response to endoplasmic reticulum stress </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0030968 </TD> <TD> endoplasmic reticulum unfolded protein response </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0006984 </TD> <TD> ER-nucleus signaling pathway </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc9"></a> </p>

<h4>antigen processing and presentation - UC.UP,CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0002474 </TD> <TD> antigen processing and presentation of peptide antigen via MHC class I </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0042590 </TD> <TD> antigen processing and presentation of exogenous peptide antigen via MHC class I </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0002479 </TD> <TD> antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0019882 </TD> <TD> antigen processing and presentation </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0048002 </TD> <TD> antigen processing and presentation of peptide antigen </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0019884 </TD> <TD> antigen processing and presentation of exogenous antigen </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002478 </TD> <TD> antigen processing and presentation of exogenous peptide antigen </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc10"></a> </p>

<h4>response to lipopolysaccharide and bacterial - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0009617 </TD> <TD> response to bacterium </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0071222 </TD> <TD> cellular response to lipopolysaccharide </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0042742 </TD> <TD> defense response to bacterium </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0031663 </TD> <TD> lipopolysaccharide-mediated signaling pathway </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0071219 </TD> <TD> cellular response to molecule of bacterial origin </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032496 </TD> <TD> response to lipopolysaccharide </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0071216 </TD> <TD> cellular response to biotic stimulus </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0009595 </TD> <TD> detection of biotic stimulus </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0016045 </TD> <TD> detection of bacterium </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002237 </TD> <TD> response to molecule of bacterial origin </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc11"></a> </p>

<h4>regulation of inflammatory response - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0002697 </TD> <TD> regulation of immune effector process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050688 </TD> <TD> regulation of defense response to virus </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002699 </TD> <TD> positive regulation of immune effector process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050690 </TD> <TD> regulation of defense response to virus by virus </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0045191 </TD> <TD> regulation of isotype switching </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002228 </TD> <TD> natural killer cell mediated immunity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002831 </TD> <TD> regulation of response to biotic stimulus </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0031348 </TD> <TD> negative regulation of defense response </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0031347 </TD> <TD> regulation of defense response </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002703 </TD> <TD> regulation of leukocyte mediated immunity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002443 </TD> <TD> leukocyte mediated immunity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0001906 </TD> <TD> cell killing </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0009615 </TD> <TD> response to virus </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050727 </TD> <TD> regulation of inflammatory response </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0001909 </TD> <TD> leukocyte mediated cytotoxicity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002449 </TD> <TD> lymphocyte mediated immunity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051607 </TD> <TD> defense response to virus </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0016064 </TD> <TD> immunoglobulin mediated immune response </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0050728 </TD> <TD> negative regulation of inflammatory response </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc12"></a> </p>

<h4>regulation of cell cycle and DNA damage response - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0030330 </TD> <TD> DNA damage response, signal transduction by p53 class mediator </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0000077 </TD> <TD> DNA damage checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0071156 </TD> <TD> regulation of cell cycle arrest </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0031570 </TD> <TD> DNA integrity checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0044774 </TD> <TD> mitotic DNA integrity checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0042770 </TD> <TD> signal transduction in response to DNA damage </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0071158 </TD> <TD> positive regulation of cell cycle arrest </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0072401 </TD> <TD> signal transduction involved in DNA integrity checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0072431 </TD> <TD> signal transduction involved in mitotic G1 DNA damage checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:2000045 </TD> <TD> regulation of G1/S transition of mitotic cell cycle </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0044783 </TD> <TD> G1 DNA damage checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:2000134 </TD> <TD> negative regulation of G1/S transition of mitotic cell cycle </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0072422 </TD> <TD> signal transduction involved in DNA damage checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0006977 </TD> <TD> DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0031571 </TD> <TD> mitotic G1 DNA damage checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0072413 </TD> <TD> signal transduction involved in mitotic cell cycle checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0044773 </TD> <TD> mitotic DNA damage checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0031572 </TD> <TD> G2 DNA damage checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0072395 </TD> <TD> signal transduction involved in cell cycle checkpoint </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc13"></a> </p>

<h4>regulation of ubiquitination and ligase activity - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0051352 </TD> <TD> negative regulation of ligase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051439 </TD> <TD> regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0031398 </TD> <TD> positive regulation of protein ubiquitination </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051443 </TD> <TD> positive regulation of ubiquitin-protein ligase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051437 </TD> <TD> positive regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0031145 </TD> <TD> anaphase-promoting complex-dependent proteasomal ubiquitin-dependent protein catabolic process </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051340 </TD> <TD> regulation of ligase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051438 </TD> <TD> regulation of ubiquitin-protein ligase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051444 </TD> <TD> negative regulation of ubiquitin-protein ligase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051351 </TD> <TD> positive regulation of ligase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051436 </TD> <TD> negative regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc14"></a> </p>

<h4>nik/nk-kappab cascade - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:1901222 </TD> <TD> regulation of NIK/NF-kappaB cascade </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0045860 </TD> <TD> positive regulation of protein kinase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0051347 </TD> <TD> positive regulation of transferase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0033674 </TD> <TD> positive regulation of kinase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0007250 </TD> <TD> activation of NF-kappaB-inducing kinase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0038061 </TD> <TD> NIK/NF-kappaB cascade </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc15"></a> </p>

<h4>regulation of ras/rac/rho gtpase activity - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0051345 </TD> <TD> positive regulation of hydrolase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032855 </TD> <TD> positive regulation of Rac GTPase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032318 </TD> <TD> regulation of Ras GTPase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032320 </TD> <TD> positive regulation of Ras GTPase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032862 </TD> <TD> activation of Rho GTPase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032321 </TD> <TD> positive regulation of Rho GTPase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032319 </TD> <TD> regulation of Rho GTPase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc16"></a> </p>

<h4>COPII vesicle coating and targeting - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0048208 </TD> <TD> COPII vesicle coating </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0048207 </TD> <TD> vesicle targeting, rough ER to cis-Golgi </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0048199 </TD> <TD> vesicle targeting, to, from or within Golgi </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0090114 </TD> <TD> COPII-coated vesicle budding </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc17"></a> </p>

<h4>negative regulation of peptidase activity - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0052547 </TD> <TD> regulation of peptidase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0010466 </TD> <TD> negative regulation of peptidase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0010951 </TD> <TD> negative regulation of endopeptidase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0052548 </TD> <TD> regulation of endopeptidase activity </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc18"></a> </p>

<h4>response to type 1 interferon - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0071357 </TD> <TD> cellular response to type I interferon </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0034340 </TD> <TD> response to type I interferon </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0060337 </TD> <TD> type I interferon-mediated signaling pathway </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc19"></a> </p>

<h4>protein N-linked glycosylation - UC.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0018196 </TD> <TD> peptidyl-asparagine modification </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0018279 </TD> <TD> protein N-linked glycosylation via asparagine </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0006487 </TD> <TD> protein N-linked glycosylation </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc20"></a> </p>

<h4>glandular cell differentiation - UC.Down</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0002068 </TD> <TD> glandular epithelial cell development </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002067 </TD> <TD> glandular epithelial cell differentiation </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0003309 </TD> <TD> type B pancreatic cell differentiation </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0003323 </TD> <TD> type B pancreatic cell development </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002066 </TD> <TD> columnar/cuboidal epithelial cell development </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0035883 </TD> <TD> enteroendocrine cell differentiation </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0002065 </TD> <TD> columnar/cuboidal epithelial cell differentiation </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc21"></a> </p>

<h4>membrane biogenesis and assembly - UC.Down</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0071709 </TD> <TD> membrane assembly </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0044091 </TD> <TD> membrane biogenesis </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc22"></a> </p>

<h4>activin receptor signaling - UC.Down</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0032925 </TD> <TD> regulation of activin receptor signaling pathway </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0032924 </TD> <TD> activin receptor signaling pathway </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc23"></a> </p>

<h4>oligodendrocyte differentiation - UC.Down,CROHNS.Down</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0048715 </TD> <TD> negative regulation of oligodendrocyte differentiation </TD> <TD> X </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0045686 </TD> <TD> negative regulation of glial cell differentiation </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0048713 </TD> <TD> regulation of oligodendrocyte differentiation </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0014014 </TD> <TD> negative regulation of gliogenesis </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0048709 </TD> <TD> oligodendrocyte differentiation </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc24"></a> </p>

<h4>cellular pattern specification - UC.Down,CROHNS.Down</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0009953 </TD> <TD> dorsal/ventral pattern formation </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0009952 </TD> <TD> anterior/posterior pattern specification </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0007389 </TD> <TD> pattern specification process </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0003002 </TD> <TD> regionalization </TD> <TD> X </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0007379 </TD> <TD> segment specification </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> </TR>
  <TR> <TD> GO:0001840 </TD> <TD> neural plate development </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> <TD>  </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc25"></a> </p>

<h4>NAD biosynthesis - CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0072525 </TD> <TD> pyridine-containing compound biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0009435 </TD> <TD> NAD biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0019363 </TD> <TD> pyridine nucleotide biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0072524 </TD> <TD> pyridine-containing compound metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0019359 </TD> <TD> nicotinamide nucleotide biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc26"></a> </p>

<h4>hormone metabolism - CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0016114 </TD> <TD> terpenoid biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0042573 </TD> <TD> retinoic acid metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0008299 </TD> <TD> isoprenoid biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0042448 </TD> <TD> progesterone metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0034754 </TD> <TD> cellular hormone metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc27"></a> </p>

<h4>response to growth hormone - CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0071378 </TD> <TD> cellular response to growth hormone stimulus </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0060416 </TD> <TD> response to growth hormone stimulus </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0060396 </TD> <TD> growth hormone receptor signaling pathway </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc28"></a> </p>

<h4>melanin metabolism - CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0006582 </TD> <TD> melanin metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0019748 </TD> <TD> secondary metabolic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0042438 </TD> <TD> melanin biosynthetic process </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a>
<a name="loc29"></a> </p>

<h4>protein dephosphorylation - CROHNS.Up</h4>

<!-- html table generated in R 3.0.1 by xtable 1.7-1 package -->

<!-- Tue Feb 11 09:58:14 2014 -->

<TABLE style="border-spacing:20px 5px;">
<TR> <TH> GOID </TH> <TH> Description </TH> <TH> UC.Down </TH> <TH> UC.Up </TH> <TH> CROHNS.Down </TH> <TH> CROHNS.Up </TH>  </TR>
  <TR> <TD> GO:0035304 </TD> <TD> regulation of protein dephosphorylation </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0035307 </TD> <TD> positive regulation of protein dephosphorylation </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
  <TR> <TD> GO:0035306 </TD> <TD> positive regulation of dephosphorylation </TD> <TD>  </TD> <TD>  </TD> <TD>  </TD> <TD> X </TD> </TR>
   </TABLE>

<p><a href="#tableLink">back</a></p>

</body>

</html>
